### PR TITLE
Bump Robolectric to 4.9

### PR DESCRIPTION
**Pull request by utzcoz**
Robolectric 4.9 was released today, see https://github.com/robolectric/robolectric/releases/tag/robolectric-4.9 for details.

### CODE REVIEW

**Comment by utzcoz** [comment]
Hi @wu-hui , could you help to take look at this PR? If there are some compatibility problems, please let me know, I can try to fix them locally. Thanks.

**Comment by utzcoz** [comment]
Oh, I found many test failures by running `./gradlew check`. I will fix them locally.

**Comment by utzcoz** [comment]
@wu-hui I keep Robolectric to 4.3.1 for transport:transport-backend-cct because I can't resolve httpclient compatibility problem quickly and properly. But other modules update Robolectric to 4.9. There are many tests depend on legacy `LooperMode`, so this PR adds LEGACY `LooperMode` for some tests to pass them.

File: firebase-database/firebase-database.gradle [hunk 1]
```diff
@@ -100,6 +100,7 @@ dependencies {
 
     testImplementation 'junit:junit:4.12'
     testImplementation 'org.mockito:mockito-core:2.25.0'
+    testImplementation 'org.codehaus.plexus:plexus-utils:3.4.2'
```
Comment (thatfiredev) [inline firebase-database/firebase-database.gradle:103]: Out of curiosity: What are we using this library for?

Comment (utzcoz) [reply firebase-database/firebase-database.gradle:103]: https://github.com/firebase/firebase-android-sdk/blob/ac3276ce7d5f5181b1c419731495eb88a98537af/firebase-database/src/test/java/com/google/firebase/database/PushIdGeneratorTest.java#L22 `PushIdGeneratorTest.java` uses `StringUtils` of this library to operate strings. It's not widely used. Old Robolectric versions has a dependency related to ant(I remember this one), and this dependency has direct/indirect dependency of `plexus-utils`. So previous `PushIdGeneratorTest.java` can use `StringUtils` of this library directly without any error. But newer Robolectric version removed this dependency related ant, so compiler throws an compiling error when I try to update Robolectric for firebse-database. This PR just declare this dependency explicitly to make compile and running pass. It's better to refactor `PushIdGeneratorTest.java` to remove `StringUtils` usage and `plexus-utils` library entirely.

**Comment by utzcoz** [comment]
Hi @rlazo @vkryachko , are there any updates for this PR? Thanks.

**Review by rlazo — APPROVED** [review]
Hi @utzcoz thanks a lot for your patience! The change looks good to me.

File: firebase-database/firebase-database.gradle (hunk 1 above)
Comment (utzcoz) [reply firebase-database/firebase-database.gradle:103]: FYI @thatfiredev I will send a new PR to remove plexus-utils.

**Pull request merged**